packages/twenty-front/src/modules/auth/sign-in-up/hooks/useSignInWithMicrosoft.ts

```diff
@@ -1,11 +1,18 @@
-import { useParams } from 'react-router-dom';
+import { useParams, useSearchParams } from 'react-router-dom';
 
 import { useAuth } from '@/auth/hooks/useAuth';
 
 export const useSignInWithMicrosoft = () => {
   const workspaceInviteHash = useParams().workspaceInviteHash;
+  const [searchParams] = useSearchParams();
```

Comment:
LGTM but I feel this could have been extracted to a hook as it's duplicated 3 times!